app/commands/user/insiders_status/update.rb

```diff
@@ -0,0 +1,62 @@
+class User::InsidersStatus::Update
+  include Mandate
+
+  queue_as :default
+
+  initialize_with :user, :status_before_unset
+
+  def call
+    eligibility_status = User::InsidersStatus::DetermineEligibilityStatus.(user)
+
+    user.with_lock do
+      case eligibility_status
+      when :eligible_lifetime
+        update_eligible_lifetime
+      when :eligible
+        update_eligible
+      when :ineligible
+        update_ineligible
+      end
+    end
+  end
```

Comment:
This isn't essential now, but I'd quite to not include the notification sending bit within the lock.

Reply:
Follow-up PR to fix this: https://github.com/exercism/website/pull/3700